src/index.js

```diff
@@ -1,50 +1,100 @@
-  console.log('Received event:', JSON.stringify(event, null, 2));
-  const bucket = event.Records[0].s3.bucket.name;
-  const key = event.Records[0].s3.object.key;
-  const getObject = (err, data) => {
-    if (err) return context.done('error', `error getting file${err}`);
-    console.log(`data:${JSON.stringify(data, null, ' ')}`);
-    try {
-      // Coverage Report Object
-      const cro = JSON.parse(data.Body);
-      if (!cro.coverage) context.done('error', `property 'coverage' not found`);
-      const coverage = Number(cro.coverage);
-      if (isNaN(coverage)) {
-        throw `property 'coverage' is not a number`;
-      }
-
-      console.log(`coverage:${coverage}!!`);
-
-      const bucket = 'zoi-public';
-
-      // generate badge
-      const badgeParam = {
-        subject: cro['subject'] || 'subject',
-        status: `${coverage}%`,
-        color: cro['color'] || 'lightgrey',
-        imageType: 'svg'
-      };
-
-      put(badgeParam, badgeParam.subject);
-    } catch (e) {
-      context.done('error', `${e}`);
-    }
-  };
-  s3.getObject({ Bucket: bucket, Key: key }, getObject);
+
+const sio = require('./shields-io');
+const readBadgeJson = require('./read-badge-json');
+
+module.exports = async function main() {
+  console.log('Ganbaru zoi!');
+  const args = getArgs();
+
+  let badgeParam;
+  try {
+    badgeParam = await readBadgeJson(args.jsonFile);
+  } catch (e) {
+    console.log('invalid json file\n', e);
+    return process.exit(1);
+  }
+
+  let badgeData;
+  try {
+    badgeData = await getBadgeData(badgeParam);
+  } catch (e) {
+    console.log('failed `getBadgeData()`\n', e);
+    return process.exit(1);
+  }
+
+  try {
+    await putToS3(badgeData, badgeParam);
+  } catch (e) {
+    console.log('failed `putToS3()`\n', e);
+    return process.exit(1);
+  }
+}
+
+function getArgs() {
+  const commandLineArgs = require('command-line-args');
+  const optionDefinitions = [
+    { name: 'jsonFile', type: String },
+    { name: 'config', type: String },
+  ];
+  try {
+    return commandLineArgs(optionDefinitions);
+  } catch (e) {
+    console.log(e);
+    process.exit(1);
+  }
+}
+
+function getBadgeData(params) {
+  return new Promise((resolve, reject) => {
+    let data = '';
+    const url = sio.url(params);
+
+    http.get(url, res => {
+      res.on('data', chunk => { data += chunk; });
+      res.on('end', () => { resolve(data); });
+      res.on('error', err => { reject(err); });
+    });
+  });
+}
+
+const contentTypeOf = imageType => {
+  switch (imageType) {
+    case 'svg':
+      return 'image/svg+xml';
+    case 'png':
+      return 'image/png';
+    default:
+      throw new Error('invalid image type.');
+  }
 };
+
+function putToS3(badgeData, badgeParam) {
+  const aws = require('aws-sdk');
+  const s3 = new aws.S3({ apiVersion: '2006-03-01' });
+  const imageType = badgeParam.imageType;
+  const contentType = contentTypeOf(imageType);
+
+  const putParam = getConfig(getArgs().config);
```

Comment:
It seems to me that `putToS3` takes [`args`](https://github.com/opt-tech/zoi/pull/7/files#diff-1fdf421c05c1140f6d71444ea2b27638R9) as a parameter.